cli/ts/utils/verifiers.ts

```diff
@@ -0,0 +1,91 @@
+import { Contract } from "ethers";
+import { genTreeProof } from "maci-crypto";
+
+/**
+ * Loop through each per vote option spent voice credits and verify it on-chain
+ *
+ * @param tallyContract The tally contract
+ * @param tallyData The tally.json file data
+ * @param voteOptionTreeDepth The vote option tree depth
+ * @param newSpentVoiceCreditsCommitment The total spent voice credits commitment
+ * @param newResultsCommitment The tally result commitment
+ *
+ * @returns list of the indexes of the tally result that failed on-chain verification
+ */
+export const verifyPerVOSpentVoiceCredits = async (
+  tallyContract: Contract,
+  tallyData: any,
+  voteOptionTreeDepth: number,
+  newSpentVoiceCreditsCommitment: bigint,
+  newResultsCommitment: bigint,
+): Promise<number[]> => {
+  const failedIndices: number[] = [];
+
+  for (let i = 0; i < tallyData.perVOSpentVoiceCredits.tally.length; i++) {
+    const proof = genTreeProof(
+      i,
+      tallyData.perVOSpentVoiceCredits.tally.map((x) => BigInt(x)),
+      voteOptionTreeDepth,
+    );
+
+    const isValid = await tallyContract.verifyPerVOSpentVoiceCredits(
+      i,
+      tallyData.perVOSpentVoiceCredits.tally[i],
+      proof,
+      tallyData.perVOSpentVoiceCredits.salt,
+      voteOptionTreeDepth,
+      newSpentVoiceCreditsCommitment,
+      newResultsCommitment,
+    );
+    if (!isValid) {
+      failedIndices.push(i);
+    }
+  }
+
+  return failedIndices;
+};
+
+/**
+ * Loop through each tally result and verify it on-chain
+ *
+ * @param tallyContract The tally contract
+ * @param tallyData The tally.json file data
+ * @param voteOptionTreeDepth The vote option tree depth
+ * @param newSpentVoiceCreditsCommitment The total spent voice credits commitment
+ * @param newPerVOSpentVoiceCreditsCommitment The per vote option voice credits commitment
+ *
+ * @returns list of the indexes of the tally result that failed on-chain verification
+ */
+export const verifyTallyResults = async (
+  tallyContract: Contract,
+  tallyData: any,
```

Comment:
we now have an interface for the tally data, please use this https://github.com/privacy-scaling-explorations/maci/blob/dev/cli/ts/utils/interfaces.ts#L25